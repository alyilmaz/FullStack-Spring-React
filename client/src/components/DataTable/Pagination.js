import React from "react";

const Pagination = ({ page, count, rowsPerPage, totalPages, setPage, setRowsPerPage }) => {
  const beginning = page === 0 ? 1: (rowsPerPage*(page)) + 1;
  const end = page ===totalPages-1 ? count : (parseInt(beginning) + parseInt(rowsPerPage) - 1);


  return (
    <>
    <div className="pagination-groups">
    <div className="pagination">
      <button data-testid="button-first" className = "page-button first" disabled={page === 0} onClick={() => setPage(0)}>First</button>
      <button data-testid="button-prev" className = "page-button prev" disabled={page === 0} onClick={() => setPage(page - 1)}>Previous</button>
      <button data-testid="button-next" className = "page-button next" disabled={page === totalPages-1} onClick={() => setPage(page + 1)}>Next</button>
      <button data-testid="button-last" className = "page-button last" disabled={page === totalPages-1} onClick={() => setPage(totalPages-1)}>Last</button>
    </div>
    <div className="page-size">
    <select required onChange={event => {setRowsPerPage(event.target.value);setPage(0);}}>
          <option data-testid="opt-1" value={5}>5</option>
          <option data-testid="opt-2" value={10}>10</option>
          <option data-testid="opt-3" value={15}>15</option>
          <option data-testid="opt-4" value={20}>20</option>
     </select>
    </div>
    <div>
    <p data-testid="text-page" >
        Page {page+1} of {totalPages}
      </p>
      </div>
      <div>
      <p data-testid="text-rows" >
        Rows: {beginning === end ? end : `${beginning} - ${end}`} of {count}
      </p>
      </div>
      </div>
    </>
  )
}

export default Pagination;