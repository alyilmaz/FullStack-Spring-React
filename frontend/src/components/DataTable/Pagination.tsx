interface Props{
    page: number,
    count: number,
    rowsPerPage: number,
    totalPages: number,
    setPage(input : number):void,
    setRowsPerPage(input : number): void
}

const Pagination = (props:Props) =>{

    const beginning:number = props.page === 0 ? 1: (props.rowsPerPage*(props.page)) + 1;
    const end = props.page ===props.totalPages-1 ? props.count : (beginning + props.rowsPerPage - 1);
  
  
    return (
      <>
      <div className="pagination-groups">
      <div className="pagination">
        <button data-testid="button-first" className = "page-button first" disabled={props.page === 0} onClick={() => props.setPage(0)}>First</button>
        <button data-testid="button-prev" className = "page-button prev" disabled={props.page === 0} onClick={() => props.setPage(props.page - 1)}>Previous</button>
        <button data-testid="button-next" className = "page-button next" disabled={props.page === props.totalPages-1} onClick={() => props.setPage(props.page + 1)}>Next</button>
        <button data-testid="button-last" className = "page-button last" disabled={props.page === props.totalPages-1} onClick={() => props.setPage(props.totalPages-1)}>Last</button>
      </div>
      <div className="page-size">
      <select id= "pageSize" required onChange={event => {props.setRowsPerPage(parseInt(event.target.value));props.setPage(0);}}>
            <option data-testid="opt-1" value={5}>5</option>
            <option data-testid="opt-2" value={10}>10</option>
            <option data-testid="opt-3" value={15}>15</option>
            <option data-testid="opt-4" value={20}>20</option>
       </select>
      </div>
      <div>
      <p id = "text-page"  >
          Page {props.page+1} of {props.totalPages}
        </p>
        </div>
        <div>
        <p id="text-rows"  >
          Rows: {beginning === end ? end : `${beginning} - ${end}`} of {props.count}
        </p>
        </div>
        </div>
      </>
    )
}

export default Pagination;