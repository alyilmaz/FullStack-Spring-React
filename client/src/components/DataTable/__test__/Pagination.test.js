/* eslint-disable testing-library/no-node-access */
import {render, screen} from "@testing-library/react";
import Pagination from "../Pagination";


test("initial button status", () =>{
    render(<Pagination page={0} count={15} rowsPerPage={5} totalPages={3} />);
    expect(screen.getByTestId("button-first")).toBeDisabled();
    expect(screen.getByTestId("button-prev")).toBeDisabled();
    expect(screen.getByTestId("button-next")).toBeEnabled();
    expect(screen.getByTestId("button-last")).toBeEnabled();
})

test("check the buttons status at end of the page", () =>{
    render(<Pagination page={2} count={15} rowsPerPage={5} totalPages={3} />);
    expect(screen.getByTestId("button-first")).toBeEnabled();
    expect(screen.getByTestId("button-prev")).toBeEnabled();
    expect(screen.getByTestId("button-next")).toBeDisabled();
    expect(screen.getByTestId("button-last")).toBeDisabled();
})

test("check the value of page and rows data", () =>{
    render(<Pagination page={0} count={15} rowsPerPage={5} totalPages={3} />);
    expect(document.getElementById("text-page").textContent).toEqual("Page 1 of 3");
    expect(document.getElementById("text-rows").textContent).toEqual("Rows: 1 - 5 of 15");
})

test("check the value of page and rows data at another page", () =>{
    render(<Pagination page={1} count={17} rowsPerPage={4} totalPages={5} />);
    expect(document.getElementById("text-page").textContent).toEqual("Page 2 of 5");
    expect(document.getElementById("text-rows").textContent).toEqual("Rows: 5 - 8 of 17");
})

test("check the all option values", () =>{
    render(<Pagination page={1} count={17} rowsPerPage={4} totalPages={5} />);
    expect(screen.getByTestId("opt-1").textContent).toEqual("5");
    expect(screen.getByTestId("opt-2").textContent).toEqual("10");
    expect(screen.getByTestId("opt-3").textContent).toEqual("15");
    expect(screen.getByTestId("opt-4").textContent).toEqual("20");
})