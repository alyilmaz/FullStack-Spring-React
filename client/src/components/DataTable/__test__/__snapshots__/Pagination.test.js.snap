// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot 1`] = `
<div
  className="pagination-groups"
>
  <div
    className="pagination"
  >
    <button
      className="page-button first"
      data-testid="button-first"
      disabled={false}
      onClick={[Function]}
    >
      First
    </button>
    <button
      className="page-button prev"
      data-testid="button-prev"
      disabled={false}
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="page-button next"
      data-testid="button-next"
      disabled={false}
      onClick={[Function]}
    >
      Next
    </button>
    <button
      className="page-button last"
      data-testid="button-last"
      disabled={false}
      onClick={[Function]}
    >
      Last
    </button>
  </div>
  <div
    className="page-size"
  >
    <select
      id="pageSize"
      onChange={[Function]}
      required={true}
    >
      <option
        data-testid="opt-1"
        value={5}
      >
        5
      </option>
      <option
        data-testid="opt-2"
        value={10}
      >
        10
      </option>
      <option
        data-testid="opt-3"
        value={15}
      >
        15
      </option>
      <option
        data-testid="opt-4"
        value={20}
      >
        20
      </option>
    </select>
  </div>
  <div>
    <p
      id="text-page"
    >
      Page 
      2
       of 
      5
    </p>
  </div>
  <div>
    <p
      id="text-rows"
    >
      Rows: 
      5 - 8
       of 
      17
    </p>
  </div>
</div>
`;
